omnidoc/lark_tree_matcher.py

```diff
@@ -0,0 +1,154 @@
+"""Tree matcher based on Lark grammar
+
+TODO: Move to lark-parser, refactor reconstruct.py to use this as base class.
+"""
+
+
+from collections import defaultdict
+
+from lark.tree import Tree
+from lark.common import ParserConf
+from lark.lexer import Token
+from lark.parsers import earley
+from lark.grammar import Rule, Terminal, NonTerminal
+
+
+def is_discarded_terminal(t):
+    return t.is_term and t.filter_out
+
+def is_iter_empty(i):
```

Comment:
if we don't want these functions to be 'private', start them with `_`. eg. `_is_iter_empty`

Reply:
do we want them to be private? I don't know if it matters.

Reply:
so in python, people use `_fn(x)` to represent private functions. [People use](https://stackoverflow.com/questions/1547145/defining-private-module-functions-in-python) `_fn` and `__fn` to *indicate* function privacy